### PR TITLE
compact: Add optional flag to disable downsampling.

### PR DESCRIPTION
**Pull request by Bplotka**
Fixes https://github.com/improbable-eng/thanos/issues/297

PTAL @drax68

### CODE REVIEW

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 	wait := cmd.Flag("wait", "Do not exit after all compactions have been processed and wait for new work.").
 		Short('w').Bool()
 
+	// TODO(bplotka): Remove this flag once https://github.com/improbable-eng/thanos/issues/297 is fixed.
+	disableDownsampling := cmd.Flag("debug.disable-downsampling", "Disables downsampling. This is not recommended " +
```
Comment (domgreen) [inline cmd/thanos/compact.go]: why is this not `compact.disable-downsampling`? Seems to be the pattern elsewhere in the codebase ... see query.

File: cmd/thanos/compact.go [hunk 2]
```diff
@@ -131,28 +138,35 @@ func runCompact(
 		if err := compactor.Compact(ctx); err != nil {
 			return errors.Wrap(err, "compaction failed")
 		}
+		level.Info(logger).Log("msg", "compaction iterations done")
 
-		// After all compactions are done, work down the downsampling backlog.
-		// We run two passes of this to ensure that the 1h downsampling is generated
-		// for 5m downsamplings created in the first run.
-		level.Info(logger).Log("msg", "start first pass of downsampling")
+		// TODO(bplotka): Remove "disableCompaction" once https://github.com/improbable-eng/thanos/issues/297 is fixed.
```
Comment (domgreen) [inline cmd/thanos/compact.go]: wrong flag `disableDownsampling`?

File: cmd/thanos/compact.go (hunk 1 above)
Comment (Bplotka) [reply cmd/thanos/compact.go]: because it's debug only and should be set only when you know consequences.

File: cmd/thanos/compact.go (hunk 2 above)
Comment (Bplotka) [reply cmd/thanos/compact.go]: nice catch ):

**Pull request merged**